core/app/models/spree/promotion/actions/create_quantity_adjustments.rb

```diff
@@ -56,10 +56,12 @@ class CreateQuantityAdjustments < CreateItemAdjustments
         # adjustment. +adjustment_amount * 3+ or $15.
         #
         def compute_amount(line_item)
+          return BigDecimal(0) if line_item.nil?
+
           adjustment_amount = calculator.compute(PartialLineItem.new(line_item))
-          if !adjustment_amount.is_a?(BigDecimal)
-            Spree::Deprecation.warn "#{calculator.class.name}#compute returned #{adjustment_amount.inspect}, it should return a BigDecimal"
-          end
+          # if !adjustment_amount.is_a?(BigDecimal)
+          #   Spree::Deprecation.warn "#{calculator.class.name}#compute returned #{adjustment_amount.inspect}, it should return a BigDecimal"
+          # end
```

Comment:
do we want to remove these lines right?

Reply:
I believe they should be kept, so that custom calculators will still generate warnings even if the ones in core don't.